orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+#include "opal/mca/event/event.h"
+
+#if OPAL_ENABLE_FT_CR == 1
+#include "orte/mca/rml/rml.h"
+#include "orte/mca/state/state.h"
+#endif
+#include "orte/mca/rml/base/base.h"
+#include "orte/mca/rml/rml_types.h"
+#include "orte/mca/routed/routed.h"
+#include "orte/mca/errmgr/errmgr.h"
+#include "orte/util/name_fns.h"
+#include "orte/runtime/orte_globals.h"
+
+#include "rml_ofi.h"
+
+static int my_priority=3;  
+static orte_rml_base_module_t* rml_ofi_component_init(int* priority);  
+static int rml_ofi_component_open(void);
+static int rml_ofi_component_close(void);
+
+
+/**
+ * component definition
+ */
+orte_rml_component_t mca_rml_ofi_component = {
+      /* First, the mca_base_component_t struct containing meta
+         information about the component itself */
+
+    .rml_version = {
+        ORTE_RML_BASE_VERSION_2_0_0,
+
+        .mca_component_name = "ofi",
+        MCA_BASE_MAKE_VERSION(component, ORTE_MAJOR_VERSION, ORTE_MINOR_VERSION,
+                              ORTE_RELEASE_VERSION),
+        .mca_open_component = rml_ofi_component_open,
+        .mca_close_component = rml_ofi_component_close,
+     },
+    .rml_data = {
+        /* The component is checkpoint ready */
+        MCA_BASE_METADATA_PARAM_CHECKPOINT
+    },
+    .rml_init = rml_ofi_component_init,
+};
+
+orte_rml_ofi_module_t orte_rml_ofi = {
+    {
+         .enable_comm = orte_rml_ofi_enable_comm,   //  [A] should we be calling this ?
+        .finalize = orte_rml_ofi_fini,
+        .query_transports = orte_rml_ofi_query_transports,
+        .send_transport_nb = orte_rml_ofi_send_transport_nb,
+        .send_buffer_transport_nb = orte_rml_ofi_send_buffer_transport_nb,
+
+    }
+};
+
+/* Local variables */
+static bool init_done = false;
+
+static int
+rml_ofi_component_open(void)
+{
+     /* Initialise endpoint and all queues */
+
+    orte_rml_ofi.fi_info_list = NULL;
+    orte_rml_ofi.min_ofi_recv_buf_sz = MIN_MULTI_BUF_SIZE;
+    orte_rml_ofi.cur_msgid = 1;
+
+    for( uint8_t conduit_id=0; conduit_id < MAX_CONDUIT ; conduit_id++) {
+        orte_rml_ofi.ofi_conduits[conduit_id].fabric =  NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].domain =  NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].av     =  NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].cq     =  NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].ep     =  NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].ep_name[0] = 0;
+        orte_rml_ofi.ofi_conduits[conduit_id].epnamelen = 0;
+        orte_rml_ofi.ofi_conduits[conduit_id].mr_multi_recv = NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].rxbuf = NULL;
+        orte_rml_ofi.ofi_conduits[conduit_id].rxbuf_size = 0;
+        orte_rml_ofi.ofi_conduits[conduit_id].progress_ev_active = false;
+    }
+
+    opal_output_verbose(1,orte_rml_base_framework.framework_output," from %s:%d rml_ofi_component_open()",__FILE__,__LINE__);
+
+    return ORTE_SUCCESS;
+}
+
+
+void free_conduit_resources( int conduit_id)
```

Comment:
This function -- and several other non-prefixed functions like it in this file -- should be static so that they don't pollute the symbol namespace.